src/core/model_config.proto

```diff
@@ -764,6 +764,86 @@ message ModelParameter
   string string_value = 1;
 }
 
+//@@
+//@@.. cpp:var:: message ModelWarmup
+//@@
+//@@   Settings used to construct the request sample for model warmup.
+//@@
+message ModelWarmup
+{
+  message InputMetaData
+  {
+    //@@  .. cpp:var:: DataType data_type
+    //@@
+    //@@     The data-type of the input.
+    //@@
+    DataType data_type = 1;
+
+    //@@    .. cpp:var:: int64 dims (repeated)
+    //@@
+    //@@       The shape of the input tensor, not including the batch dimension.
+    //@@
+    repeated int64 dims = 2;
+
+    //@@  .. cpp:var:: oneof input_data_type
+    //@@
+    //@@     Specify how the input data is generated. If the input has STRING
+    //@@     data type and 'random_data' is set, the data generation will fall
+    //@@     back to 'zero_data'.
+    //@@
+    oneof input_data_type
+    {
+      //@@
+      //@@  .. cpp:var:: bool zero_data
+      //@@
+      //@@     The identifier for using zeros as input data. Note that the
+      //@@     field data will not be checked, instead, zero data will be used
```

Comment:
What does "field data" mean?

Reply:
I should have just said "value of 'zero_data'"